WireDomain/Sources/WireDomain/Repositories/Team/TeamRepository.swift

```diff
@@ -129,6 +135,56 @@ final class TeamRepository: TeamRepositoryProtocol {
         }
     }
 
+    public func pullSelfTeamRoles() async throws {
+        let teamRoles = try await fetchSelfTeamRolesRemotely()
+        try await storeTeamRolesLocally(teamRoles)
+    }
+
+    public func pullSelfTeamMembers() async throws {
+        let teamMembers = try await fetchSelfTeamMembersRemotely()
+        try await storeTeamMembersLocally(teamMembers)
+    }
+
+    public func pullSelfLegalHoldStatus() async throws {
+        let selfUser = await context.perform { [userRepository] in
+            userRepository.fetchSelfUser()
+        }
+
+        let selfUserLegalHold = try await fetchSelfLegalhold()
+
+        switch selfUserLegalHold.status {
+        case .pending:
+            guard let selfClientID = selfUser.selfClient()?.remoteIdentifier else {
+                return
+            }
+
+            await userRepository.addLegalHoldRequest(
+                for: selfUser.remoteIdentifier,
```

Comment:
accessing remoteIdentifier outside of the context isn't a CoreData threading violation?

Reply:
it is. fixed in https://github.com/wireapp/wire-ios/pull/1999/commits/557391b8308dabd6c0db318885ad14a8cba7ef66